### ISSUE TITLE
asyncio implementation?

### ISSUE THREAD

**Issue by asthypar** [SolidElectronics/evl-emu#7]
This is not technically an issue, but because this is such a great project and has been the backbone of my HA deployment from a security perspective for the last few years, I can help but continue to re-read this section and not feel that it's one of the only things missing:

"Ideally this would be rewritten in a cleaner way with asyncio but that’s currently beyond my ability."

Unfortunately, this is also beyond my ability, but perhaps some of the other contributors or watchers could tackle this and knock it out quickly and just put one of the final touches on something that has already been so functional and stable?

I am happy to provide support from a testing perspective (running debugging, collecting logs, testing functionality) but unfortunately my dev skills do not extend to the full range required for something like this.

**Comment by cbschuld** [comment]
@asthypar I am not sure there would be a realized benefit for that type of update candidly.  Sure, it would probably be better using an async model for readability in the source but I am not sure there is any specific benefit as long as this solution is working for you then I would suggest it's fine.  Switching the style from multi threaded to more of an async approach is just a way changing where the async nature of this solution is addressed.  I think mainly Paul is saying here (@SolidElectronics ) that "hey, there is possibly a better/cleaner way to write/present this solution using async but it's not in my wheelhouse so here's my solution."

TLDR;  if it is working for you don't touch it or worry about it
